src/plugins/orangehrmRecruitmentPlugin/test/Controller/ApplicantControllerTest.php

```diff
@@ -0,0 +1,138 @@
+     * @return void
+     */
+    public function testHandle(
+        int $vacancyId,
+        string $firstName,
+        string $middleName,
+        string $lastName,
+        string $email,
+        string $contactNumber,
+        string $consentToKeepData
+    ): void {
+        $q = $this->createQueryBuilder(Candidate::class, 'candidate');
+        $q->leftJoin('candidate.candidateVacancy', 'candidateVacancy');
+        $q->leftJoin('candidateVacancy.vacancy', 'vacancy');
+
+        $this->assertEquals(5, $this->getPaginator($q)->count());
+        $this->createKernelWithMockServices([
+            Services::CANDIDATE_SERVICE => new CandidateService(),
+            Services::RECRUITMENT_ATTACHMENT_SERVICE => new RecruitmentAttachmentService(),
+            Services::DATETIME_HELPER_SERVICE => new DateTimeHelperService(),
+            Services::NORMALIZER_SERVICE => new NormalizerService(),
+            Services::CONFIG_SERVICE => new ConfigService(),
+        ]);
+        $csrfTokenManager = new CsrfTokenManager();
+        $token = $csrfTokenManager->getToken('recruitment-applicant')->getValue();
+        $pathToResume = __DIR__.'/../fixtures/resume.txt';
+        $uploadedFile = new UploadedFile($pathToResume, 'resume.txt', 'text/plain', null, true);
+
+        $applicantController = $this->getMockBuilder(ApplicantController::class)
+            ->onlyMethods(['forward'])
+            ->getMock();
+        $applicantController->method('forward')
+            ->willReturnCallback(function ($forwardedController, $attributes) {
+                return new Response(
+
+                );
+            });
+
+        $request = $this->getHttpRequest([], [
+            '_token' => $token,
+            'vacancyId' => $vacancyId,
+            'firstName' => $firstName,
+            'middleName' => $middleName,
+            'lastName' => $lastName,
+            'email' => $email,
+            'contactNumber' => $contactNumber,
+            'consentToKeepData' => $consentToKeepData,
+        ], [], [], [
+            'resume' => $uploadedFile,
+        ]);
+        $response = $applicantController->handle($request);
+        $this->assertEquals(6, $this->getPaginator($q)->count());
+        $this->assertEquals('Peter', $q->select('candidate.firstName')->where('candidate.id = 1')->getQuery()->execute()[0]['firstName']);
+        $this->assertNotEquals('Zaman', $q->select('candidate.lastName')->where('candidate.id = 3')->getQuery()->execute()[0]['lastName']);
+        $this->assertEquals('Senior Technical Supervisor', $q->select('vacancy.name')->where('vacancy.id = 2')->getQuery()->execute()[0]['name']);
+        $this->assertEquals('APPLICATION INITIATED', $q->select('candidateVacancy.status')->where('candidateVacancy.id = 1')->getQuery()->execute()[0]['status']);
+    }
+
+    /**
+     * @return array[]
+     */
+    public function dataProviderForController(): array
+    {
+        return [
+            [
```

Comment:
usually data providers used to handle multiple test cases. seems handle only one here

Reply:
added two